packages/salesforce-adapter/src/filters/changed_at_singleton.ts

```diff
@@ -0,0 +1,73 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  CORE_ANNOTATIONS,
+  Element,
+  ElemID,
+  InstanceElement,
+  isInstanceElement,
+  Values,
+} from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import _ from 'lodash'
+import { FilterWith, LocalFilterCreator } from '../filter'
+import { isMetadataInstanceElement } from '../transformers/transformer'
+import { ArtificialTypes, INSTANCE_FULL_NAME_FIELD } from '../constants'
+import { getChangedAtSingleton, safeApiName } from './utils'
+
+const { groupByAsync, awu } = collections.asynciterable
+
+
+const createChangedAtSingletonInstanceValues = (metadataInstancesByType: Record<string, InstanceElement[]>): Values => {
+  const instanceValues: Values = {}
+  Object.entries(metadataInstancesByType).forEach(([metadataType, instances]) => {
+    instanceValues[metadataType] = {}
+    instances.forEach(instance => {
+      const instanceName = instance.value[INSTANCE_FULL_NAME_FIELD]
+      instanceValues[metadataType][instanceName] = instance.annotations[CORE_ANNOTATIONS.CHANGED_AT]
+    })
+  })
+  return instanceValues
+}
```

Comment:
why use stateful code for a simple transformation?

```suggestion
const createChangedAtSingletonInstanceValues = (metadataInstancesByType: Record<string, InstanceElement[]>): Values => _.mapValues(
  metadataInstancesByType,
  instances => Object.fromEntries(
    instances.map(instance => [
      instance[INSTANCE_FULL_NAME_FIELD],
      instance.annotations[CORE_ANNOTATIONS.CHANGED_AT],
    ])
  )
)
```

also, is there a reason we don't have a type for the value?
I'd expect that we would have a type of it given that we need to use it in multiple places and I assume we rely on its structure in the code that needs to decide what to fetch

Reply:
What would you expect this type to be? `Record<string, Record<string,string>>`? I personally don't think it adds anything, as we simply use **_.get(instanceValue, typeName, instanceName)**

Reply:
you can say that about any type...
we want to have types because they make it easier to know what we are looking at, we want to know that when we `_.get(value, typeName, instanceName)` we will get back a string and not a list or an object or nothing. that is the whole point of writing typescript and not plain javascript (plus, not sure why we would need to use `_.get` to access a value in an object).

`Values` is basically `any`, and we want to minimize our use of `any` because it removes the value we get from using typescript and not writing in plain javascript...